apps/gnome-mobile/src/app/(app)/addfriend.tsx

```diff
@@ -1,113 +1,236 @@
+import { BottomSheetModal } from "@gorhom/bottom-sheet";
+import Clipboard from "@react-native-clipboard/clipboard";
 import { useNavigation } from "@react-navigation/native";
 import { useRouter } from "expo-router";
-import React, { useEffect, useState } from "react";
-import {
-  FlatList,
-  Image,
-  TextInput,
-  TouchableOpacity,
-  View,
-} from "react-native";
-import { SafeAreaView } from "react-native-safe-area-context";
+import React, { useEffect, useRef, useState } from "react";
+import { Image, TouchableOpacity, View } from "react-native";
+import { QrCodeSvg } from "react-native-qr-svg";
+import { useCameraDevice } from "react-native-vision-camera";
 import ArrowLeft from "@/assets/icons/arrow-left.svg";
-import BackIcon from "@/assets/icons/arrow-left.svg";
-import SearchIcon from "@/assets/icons/search.svg";
+import CameraIcon from "@/assets/icons/camera.svg";
+import CopyIcon from "@/assets/icons/copy.svg";
+import PlusIcon from "@/assets/icons/plus.svg";
+import RefreshIcon from "@/assets/icons/refresh.svg";
+import ShareIcon from "@/assets/icons/share-right.svg";
+import { ConfirmDialog } from "@/components/ConfirmDialog";
+import { Scanner } from "@/components/Scanner";
+import { Input } from "@/components/ui/input";
 import { Text } from "@/components/ui/text";
+import { cn } from "@/lib/utils";
+import { useAuthStore } from "@/store/useAuthStore";
+import { useFriendsStore } from "@/store/useFriendsStore";
 
-type FriendsListProps = {
-  friends: { id: string; name: string; avatar?: string }[];
-};
+type DialogActions = "regenerate-invite-code" | "scan-qr-code";
 
-const FriendsList: React.FC<FriendsListProps> = ({ friends }) => {
+export default function AddFriendScreen() {
   const navigation = useNavigation();
   const router = useRouter();
-  //Header
-  useEffect(() => {
-    navigation.setOptions({
-      headerLeft: () => (
-        <TouchableOpacity className="p-5" onPress={() => router.back()}>
-          <BackIcon className="w-7 h-7" />
-        </TouchableOpacity>
-      ),
-      headerTitle: () => (
-        <Text className="text-white text-2xl text-center font-bold tracking-wide">
-          Dodaj znajomego
-        </Text>
-      ),
-      headerTitleAlign: "center",
-      headerShadowVisible: false,
-      headerShown: true,
+  const { user, regenerateInviteCode } = useAuthStore();
+  const { addFriend } = useFriendsStore();
+  const bottomSheetModalRef = useRef<BottomSheetModal>(null);
```

Comment:
rozdzieliłbym to na dwa modale

(i dialogaction do usunięcia)

nazwy typu:
regenerateInviteCodeSheetRef
scanInvitationSheetRef